### PR TITLE
Sub text modifiers

### PR DESCRIPTION
**Pull request by boiscljo**
Allow modifier to be applied to only parts of the text

Support for SPECIAL_COLOR inside the string but won't apply on the whole string
unknown

### CODE REVIEW

File: src/main/java/com/iridium/iridiumcolorapi/IridiumColorAPI.java [hunk 1]
```diff
@@ -34,7 +35,8 @@
      */
     private static final boolean SUPPORTS_RGB = VERSION >= 16;
 
-    private static final List<String> SPECIAL_COLORS = Arrays.asList("&l", "&n", "&o", "&k", "&m","§l", "§n", "§o", "§k", "§m");
+    private static final List<String> SPECIAL_COLORS = Arrays.asList("&l", "&n", "&o", "&k", "&m", "§l", "§n", "§o",
```
Comment (PeachesMLG) [inline src/main/java/com/iridium/iridiumcolorapi/IridiumColorAPI.java:38]: why'd you add the new line?

File: src/main/java/com/iridium/iridiumcolorapi/IridiumColorAPI.java [hunk 2]
```diff
@@ -64,7 +66,8 @@
      *
      * @since 1.0.2
      */
-    private static final List<Pattern> PATTERNS = Arrays.asList(new GradientPattern(), new SolidPattern(), new RainbowPattern());
+    private static final List<Pattern> PATTERNS = Arrays.asList(new GradientPattern(), new SolidPattern(),
```
Comment (PeachesMLG) [inline src/main/java/com/iridium/iridiumcolorapi/IridiumColorAPI.java:69]: Remove this newline aswell

File: src/main/java/com/iridium/iridiumcolorapi/IridiumColorAPI.java [hunk 3]
```diff
@@ -234,18 +250,18 @@ public static String stripColorFormatting(@Nonnull String string) {
         };
 
         for (int i = 0; i < step; i++) {
-            Color color = new Color(start.getRed() + ((stepR * i) * direction[0]), start.getGreen() + ((stepG * i) * direction[1]), start.getBlue() + ((stepB * i) * direction[2]));
+            Color color = new Color(start.getRed() + ((stepR * i) * direction[0]),
```
Comment (PeachesMLG) [inline src/main/java/com/iridium/iridiumcolorapi/IridiumColorAPI.java:253]: and this one

File: src/main/java/com/iridium/iridiumcolorapi/IridiumColorAPI.java [hunk 4]
```diff
@@ -258,7 +274,9 @@ private static ChatColor getClosestColor(Color color) {
         double nearestDistance = Integer.MAX_VALUE;
 
         for (Color constantColor : COLORS.keySet()) {
-            double distance = Math.pow(color.getRed() - constantColor.getRed(), 2) + Math.pow(color.getGreen() - constantColor.getGreen(), 2) + Math.pow(color.getBlue() - constantColor.getBlue(), 2);
+            double distance = Math.pow(color.getRed() - constantColor.getRed(), 2)
```
Comment (PeachesMLG) [inline src/main/java/com/iridium/iridiumcolorapi/IridiumColorAPI.java:277]: and this

File: src/main/java/com/iridium/iridiumcolorapi/IridiumColorAPI.java [hunk 5]
```diff
@@ -290,7 +308,8 @@ private static int getVersion() {
 
         // 1.13.2, 1.14.4, etc...
         int lastDot = version.lastIndexOf('.');
-        if (version.indexOf('.') != lastDot) version = version.substring(0, lastDot);
+        if (version.indexOf('.') != lastDot)
```
Comment (PeachesMLG) [inline src/main/java/com/iridium/iridiumcolorapi/IridiumColorAPI.java:311]: and this

**Review by PeachesMLG — CHANGES_REQUESTED** [review]
Remove the new Lines

**Review by PeachesMLG — CHANGES_REQUESTED** [review]
Would also be good to get some unit tests around the new changes

**Comment by boiscljo** [comment]
Sorry that I accidently did a "Format document" in my IDE, that's why all the new lines.

File: src/main/java/com/iridium/iridiumcolorapi/IridiumColorAPI.java (hunk 1 above)
Comment (boiscljo) [reply src/main/java/com/iridium/iridiumcolorapi/IridiumColorAPI.java]: IDE autoformatting, sorry, I fixed it in the new commit

**Comment by PeachesMLG** [comment]
Also add a unit test for gradients aswell

**Comment by boiscljo** [comment]
Should I fix the 2 Codacy Static Code Analysis?
I added the UnitTests

**Comment by PeachesMLG** [comment]
> Should I fix the 2 Codacy Static Code Analysis? I added the UnitTests

Yeah fix those aswell

**Pull request merged**